### ISSUE TITLE
Feature request: Multiple databases

### ISSUE THREAD

**Issue by wolfgangjt** [jpxthu/EPGP-Classic#48]
I like a few guilds are splitting EPGP for BWL/MC and AQ40. Would there be a possibility to have multiple databases and be able to swap between them easily?

**Comment by jpxthu** [comment]
You can use `import/export` in the `log` frame. It contains EP/GP data. You can save them info a local text file.

Actually I have thought about multiple databases by adding extra lines into officer’s note, but may not compatible with the origin format. You know, a lot users will complain about breaking changes.

I am thinking about create a new addon named `EPGP Advanced`, base on newest `EPGP` and add some breaking changes.

**Comment by wolfgangjt** [comment]
I am guessing when you import data it overwrites the officer notes with the imported data? Just making sure we dont wipe out existing data!

And I am always up for new cool features.

**Comment by jpxthu** [comment]
Yes. `Import` will overwrite the officer's notes. Be careful. By the way, only `export` data could be imported, `export detail` is unable to import for now.

Ha-ha. I have heavy works to do, too. Don't be too tired.

**Comment by marknl** [comment]
I could see a solution where you trigger import/export functionality on "database switching".

The exports can be locally saved as a json blob into a saved variable array (for multiple db's) and imported when you switch db's.

Basically back-upping and restoring snapshots of the officer notes would be a start.

**Comment by trumpetx** [comment]
This is something we were looking at doing as well.  I looked at the code to see if it would be possible to hack in and as @jpxthu mentioned, it would have breaking changes.  I was considering doing something like a "context switch" where you'd have raid "A, B, C" etc.  Then the officer note would be : `A:12345,1234;B:54321,5432`  Then you'd just "context switch" at the beginning of the raid or during an export.  But all that is a lot of refactoring to do for maintaining backwards compatibility.  Then you'd have to figure out if the contexts supported different values, different weights, etc -- that's when it would get tricky for sure.
